openwisp_monitoring/device/schema.py

```diff
@@ -142,6 +142,23 @@
                             "tx_carrier_errors": {"type": "integer"},
                         },
                     },
+                    "addresses": {
+                        "type": "array",
+                        "title": "Addresses",
+                        "uniqueItems": True,
+                        "additionalItems": True,
+                        "items": {
+                            "additionalProperties": True,
+                            "title": "Address",
+                            "type": "object",
+                            "properties": {
+                                "proto": {"type": "string"},
+                                "family": {"type": "string"},
+                                "address": {"type": "string", "format": "email"},
```

Comment:
Is it a mistake?

```
                    "ip_address": {
                        "type": "string",
                        "anyOf": [{"format": "ipv4"}, {"format": "ipv6"}],
                    },
```

should work better

Reply:
Well, I checked up all formats and nothing was validating so I was figuring it out then. Now it validates, just read the jsonschema docs again and found out `format_checker` isn't enabled by default, changed it and it validates all formats perfectly. 
Thanks, for the suggestion. Any reason why you prefer `anyOf` over `oneOf`, I haven't used **jsonschema** much, hence curious :smiley:

Reply:
For this case the result is the same so it doesn't matter. Probably anyOf is faster because of short-circuit evaluation.